src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -21,6 +34,235 @@ private SslStreamCertificateContext(X509Certificate2 target, X509Certificate2[]
+                GetOcspResponseAsync().AsTask();
+            }
+        }
+
+        internal byte[]? GetOcspResponseNoWaiting()
+        {
+            try
+            {
+                ValueTask<byte[]?> task = GetOcspResponseAsync();
+
+                if (task.IsCompletedSuccessfully)
+                {
+                    return task.Result;
+                }
+            }
+            catch
+            {
+            }
+
+            return null;
+        }
+
+        internal ValueTask<byte[]?> GetOcspResponseAsync()
+        {
+            if (_staplingForbidden)
+            {
+                return ValueTask.FromResult((byte[]?)null);
+            }
+
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+
+            if (now > _ocspExpiration)
+            {
+                return DownloadOcsp();
+            }
+
+            if (now > _nextDownload)
+            {
+                // Calling DownloadOcsp will activate a Task to initiate
+                // in the background.  Further calls will attach to the
+                // same Task if it's still running.
+                //
+                // We don't want the result here, just the task to background.
+#pragma warning disable CA2012 // Use ValueTasks correctly
+                DownloadOcsp();
+#pragma warning restore CA2012 // Use ValueTasks correctly
+            }
+
+            return ValueTask.FromResult(_ocspResponse);
+        }
+
+        private ValueTask<byte[]?> DownloadOcsp()
+        {
+            Task<byte[]?>? pending = _pendingDownload;
+
+            if (pending is not null && !pending.IsFaulted)
+            {
+                return new ValueTask<byte[]?>(pending);
+            }
+
+            if (_ocspUrls is null && _ca is not null)
+            {
+                foreach (X509Extension ext in Certificate.Extensions)
+                {
+                    if (ext is X509AuthorityInformationAccessExtension aia)
+                    {
+                        foreach (string entry in aia.EnumerateOcspUris())
+                        {
+                            if (Uri.TryCreate(entry, UriKind.Absolute, out Uri? uri))
+                            {
+                                if (uri.Scheme == UriScheme.Http)
+                                {
+                                    (_ocspUrls ??= new List<string>()).Add(entry);
+                                }
+                            }
+                        }
+
+                        break;
+                    }
+                }
+            }
+
+            if (_ocspUrls is null)
+            {
+                _ocspExpiration = _nextDownload = DateTimeOffset.MaxValue;
+                return new ValueTask<byte[]?>((byte[]?)null);
+            }
+
+            lock (SslContexts)
+            {
+                pending = _pendingDownload;
+
+                if (pending is null || pending.IsFaulted)
```

Comment:
If it's faulted, will someone somewhere have accessed its Exception or otherwise observed what the error was?